tabular/src/autogluon/tabular/models/text_prediction/text_prediction_v1_model.py

```diff
@@ -0,0 +1,258 @@
+            pd.DataFrame({self._label_column_name: pd.concat([y_train, y_val])}),
+            metadata=None,
+            label_columns=None,
+            provided_column_properties=None
+        )
+        column_properties = collections.OrderedDict(list(feature_column_properties.items()) +
+                                                    list(label_column_property.items()))
+        problem_type, label_shape = infer_problem_type(column_properties=column_properties,
+                                                       label_col_name=self._label_column_name)
+        eval_metric, stopping_metric, log_metrics =\
+            infer_eval_stop_log_metrics(problem_type,
+                                        label_shape=label_shape,
+                                        eval_metric=self.eval_metric,
+                                        stopping_metric=self.stopping_metric)
+        search_space = hyperparameters['models']['BertForTextPredictionBasic']['search_space']
+        self.model = BertForTextPredictionBasic(column_properties=column_properties,
+                                                feature_columns=list(X_train.columns),
+                                                label_columns=[self._label_column_name],
+                                                problem_types=[problem_type],
+                                                label_shapes=[label_shape],
+                                                stopping_metric=stopping_metric,
+                                                log_metrics=log_metrics,
+                                                output_directory=os.path.join(self.path, self.name),
+                                                logger=logger,
+                                                base_config=None,
+                                                search_space=search_space)
+        return column_properties
+
+    def _get_default_auxiliary_params(self) -> dict:
+        default_auxiliary_params = super()._get_default_auxiliary_params()
+        extra_auxiliary_params = dict(
+            get_features_kwargs=dict(
+                valid_raw_types=[R_INT, R_FLOAT, R_OBJECT],
+                invalid_special_types=[S_TEXT_NGRAM, S_TEXT_AS_CATEGORY, S_TEXT_SPECIAL],
+            ),
+        )
+        default_auxiliary_params.update(extra_auxiliary_params)
+        print('In TextPrediction, default_auxiliary_params=', default_auxiliary_params)
+        return default_auxiliary_params
+
+    def _set_default_params(self):
+        try:
+            from autogluon.text.text_prediction.dataset import TabularDataset
+            from autogluon.text.text_prediction.text_prediction import ag_text_prediction_params
+        except ImportError:
+            raise ImportError(AG_TEXT_IMPORT_ERROR)
+        super()._set_default_params()
+        self.params = ag_text_prediction_params.create('default')
+
+    def _fit(self, X_train: pd.DataFrame, y_train: pd.Series,
```

Comment:
This will currently crash if X_val, y_val are None, you can test this by adding `refit_full=True` to task.fit args. Can you update the code to properly handle this case?

Additionally, for refit_full to work properly, the model needs to be trained with the same number of epochs and same hyperparameters as it was for the final model prior to refit_full.

As an example, in CatBoost at the end of `_fit`:

```
self.params_trained['iterations'] = self.model.tree_count_
```

This ensures that the refit_full model sets the hyperparameter `self.params['iterations'] == self.model.tree_count_`

Reply:
This comment still applies.